tsdb/wal/wal.go

```diff
@@ -731,32 +731,27 @@ func (w *WAL) Close() (err error) {
 		return errors.New("wal already closed")
 	}
 
-	if w.segment == nil {
-		w.closed = true
-		return nil
-	}
+	donec := make(chan struct{})
+	w.stopc <- donec
+	<-donec
 
 	// Flush the last page and zero out all its remaining size.
 	// We must not flush an empty page as it would falsely signal
 	// the segment is done if we start writing to it again after opening.
 	if w.page.alloc > 0 {
-		if err := w.flushPage(true); err != nil {
-			return err
+		if err = w.flushPage(true); err != nil {
+			level.Error(w.logger).Log("msg", "flush current page", "err", err)
 		}
 	}
 
-	donec := make(chan struct{})
-	w.stopc <- donec
-	<-donec
-
 	if err = w.fsync(w.segment); err != nil {
 		level.Error(w.logger).Log("msg", "sync previous segment", "err", err)
 	}
-	if err := w.segment.Close(); err != nil {
+	if err = w.segment.Close(); err != nil {
```

Comment:
Please use `:`  - we actually want to start local error and make sure it never escapes. It's easy to forget to not add `:` that's why passing error like this is non trivial. 

Please instead let's use `tsdb_errors.MultiError` which will add all potential errors nicely. (:

Reply:
ok. Use `err =` to record error during closing and return the last err. As mentioned above, it's better to use `tsdb_errors.MultiError` to record errors.